app/models/course/assessment/tab.rb

```diff
@@ -27,4 +30,15 @@ def validate_before_destroy
     errors.add(:base, :deletion) unless safe
     safe
   end
+
+  # Reassign the assessment folders to new category if the category changed.
+  def reassign_folders
+    # Category association might not be updated when category_id changed
+    new_parent_folder = Course::Assessment::Category.find(category_id).folder
+
+    folders.each do |folder|
+      folder.parent = new_parent_folder
+      return false unless folder.save
```

Comment:
Random qn: just to check, the update of the folders are all executed in 1 transaction (instead of separate ones right)? Since this is a callback which is part of other changes.